components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -711,3 +800,44 @@ func mergeInfringingWorkspaces(vws []InfringingWorkspace) (vw InfringingWorkspac
 func (agent *Smith) RegisterMetrics(reg prometheus.Registerer) error {
 	return agent.metrics.Register(reg)
 }
+
+func (agent *Smith) checkEgressTrafficCallback(pid int, pidCreationTime time.Time) (*Infringement, error) {
```

Comment:
A unit test for `checkEgressTrafficCallback` would be awesome!

Reply:
Was a bit tricky because of the dependency with time and with proc but this is addressed in this patch https://github.com/gitpod-io/gitpod/commit/03094bcfe64307cf410a674fd859b0e7a4cb5134.patch